### PR TITLE
peazip: new, 7.7.0

### PR DESCRIPTION
**Pull request by Fearyncess**
Topic Description
-----------------
Introduce `peazip`, version 7.7.0

Package(s) Affected
-------------------
* `peazip`

Security Update?
----------------
No

Architectural Progress
----------------------
- [x] AMD64 `amd64`   
- [x] AArch64 `arm64`

----

After the pull request is merged, all package(s) affected must be rebuilt against the `stable` Git tree and environment (only `stable` repository should be enabled in `sources.list`). This section marks the progress above.

Please, make sure the list of architectures below matches the ones above.

Post-Merge Architectural Progress
---------------------------------

- [ ] AMD64 `amd64`   
- [ ] AArch64 `arm64`

### CODE REVIEW

File: extra-utils/peazip/autobuild/build [hunk 1]
```diff
@@ -0,0 +1,52 @@
+if   [[ "${CROSS:-$ARCH}" = "amd64" ]]; then
+    ARCH_OPTS="x86_64"
+elif [[ "${CROSS:-$ARCH}" = "arm64" ]]; then
+    ARCH_OPTS="aarch64"
+fi
+
```
Comment (MingcongBai) [inline extra-utils/peazip/autobuild/build:7]: With Lazarus *

File: extra-utils/peazip/autobuild/build [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+abinfo "Building PeaZip by Lazarus..."
+lazbuild \
+  --verbose \
+  --lazarusdir=/usr/share/lazarus \
+  --cpu=${ARCH_OPTS} \
+  --widgetset=gtk2 \
+  --max-process-count=$(nproc --all) \
+  -B project_pea.lpi project_peach.lpi
```
Comment (MingcongBai) [inline extra-utils/peazip/autobuild/build]: Use absolute path here.

File: extra-utils/peazip/autobuild/build [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+
+abinfo "Installing PeaZip..."
+mkdir -v "${PKGDIR}/usr/bin" -p
```
Comment (MingcongBai) [inline extra-utils/peazip/autobuild/build]: Use install -D where possible. This is unnecessarily long.

File: extra-utils/peazip/autobuild/build [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+mkdir -v "${PKGDIR}/usr/share/kservices5/ServiceMenus" -p
+
+install -v -m755 "${SRCDIR}/peazip" "${PKGDIR}/usr/lib/peazip/peazip"
+install -v -m755 "${SRCDIR}/pea" "${PKGDIR}/usr/lib/peazip/pea"
+install -v -m644 "${SRCDIR}/../peazip_help-${PKGVER}.pdf" "${PKGDIR}/usr/share/doc/peazip/peazip_help.pdf"
+install -v -m644 "${SRCDIR}/FreeDesktop_integration/peazip.png" "${PKGDIR}/usr/share/pixmaps/peazip.png"
+install -v -m644 "${SRCDIR}/FreeDesktop_integration/peazip.desktop" "${PKGDIR}/usr/share/applications/peazip.desktop"
+
+pushd "${PKGDIR}/usr/bin"
+ln -sv /usr/lib/peazip/peazip ./peazip
```
Comment (MingcongBai) [inline extra-utils/peazip/autobuild/build]: Symlink target should always be relative.

File: extra-utils/peazip/autobuild/build [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+ln -sv /usr/lib/peazip/pea ./pea
+popd
+
+pushd "${PKGDIR}/usr/lib/peazip"
+for i in 7z brotli zpaq zstd upx unrar unace
+do
+  mkdir -v "${PKGDIR}/usr/lib/peazip/res/$i"
+  ln -sv /usr/bin/$i ./res/$i/$i
```
Comment (MingcongBai) [inline extra-utils/peazip/autobuild/build]: Same as above.

File: extra-utils/peazip/autobuild/build [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+done
+popd
+
+abinfo "Installing PeaZip l10n supports..."
```
Comment (MingcongBai) [inline extra-utils/peazip/autobuild/build]: L10n data *

File: extra-utils/peazip/autobuild/build [hunk 7]
```diff
@@ -0,0 +1,52 @@
...
+pushd "${SRCDIR}/../peazip-${PKGVER}.about_translations/lang"
+install -v -m644 *.txt "${PKGDIR}/usr/lib/peazip/res/lang"
```
Comment (MingcongBai) [inline extra-utils/peazip/autobuild/build]: Use absolute path here.

File: extra-utils/peazip/autobuild/build [hunk 8]
```diff
@@ -0,0 +1,52 @@
...
+popd
+
+abinfo "Installing KDE's Dolphin Integrations..."
+pushd "${SRCDIR}/FreeDesktop_integration/kde4-dolphin/usr/share/kde4/services/ServiceMenus"
+install -v -m644 *.desktop "${PKGDIR}/usr/share/kservices5/ServiceMenus"
```
Comment (MingcongBai) [inline extra-utils/peazip/autobuild/build]: Use absolute path here.

File: extra-utils/peazip/autobuild/build [hunk 9]
```diff
@@ -0,0 +1,52 @@
+if   [[ "${CROSS:-$ARCH}" = "amd64" ]]; then
```
Comment (MingcongBai) [inline extra-utils/peazip/autobuild/build]: Did you reference this build script from anywhere? If so, please indicate the origin(s).

File: extra-utils/peazip/autobuild/defines [hunk 10]
```diff
@@ -0,0 +1,7 @@
+PKGNAME=peazip
+PKGSEC=utils
+PKGDES="Cross-platform file and archive manager written in Pascal"
```
Comment (MingcongBai) [inline extra-utils/peazip/autobuild/defines]: No one cares about what language is written in. Leave out unnecessary detail in PKGDES.

File: extra-utils/peazip/autobuild/patch [hunk 11]
```diff
@@ -0,0 +1,9 @@
+abinfo Converting files for patching...
+dos2unix "${SRCDIR}/rmd160.pas" "${SRCDIR}/tsc.pas" "${SRCDIR}/peach.pas"
+pushd ${SRCDIR}
+abinfo Applying the patches...
```
Comment (MingcongBai) [inline extra-utils/peazip/autobuild/patch]: Please move abinfo inside for loop, and indicate which patch is being applied.

File: extra-utils/peazip/autobuild/patches/0003-make-peazip-use-system's-freeware-version-unrar.patch [hunk 12]
```diff
@@ -0,0 +1,11 @@
+--- a/peach.pas	2021-01-15 15:04:15.869899477 +0800
```
Comment (MingcongBai) [inline extra-utils/peazip/autobuild/patches/0003-make-peazip-use-system's-freeware-version-unrar.patch]: Please do not use ‘ symbol in file name.

**Review by MingcongBai — CHANGES_REQUESTED** [review]

File: extra-utils/peazip/autobuild/build (hunk 5 above)
Comment (Fearyncess) [reply extra-utils/peazip/autobuild/build]: It's use system's exec binary. I think it should be absolute path.

Comment (MingcongBai) [reply extra-utils/peazip/autobuild/build]: This is irrelevant. If you set this as absolute path, then when running the binary in a non-current sysroot would result in possible file-not-found situations.

File: extra-utils/peazip/autobuild/build [hunk 13]
```diff
@@ -0,0 +1,43 @@
...
+
+install -v -Dm755 "${SRCDIR}/peazip" "${PKGDIR}/usr/lib/peazip/peazip"
+install -v -Dm755 "${SRCDIR}/pea" "${PKGDIR}/usr/lib/peazip/pea"
+install -v -Dm644 "${SRCDIR}/../peazip_help-${PKGVER}.pdf" "${PKGDIR}/usr/share/doc/peazip/peazip_help.pdf"
+install -v -Dm644 "${SRCDIR}/FreeDesktop_integration/peazip.png" "${PKGDIR}/usr/share/pixmaps/peazip.png"
+install -v -Dm644 "${SRCDIR}/FreeDesktop_integration/peazip.desktop" "${PKGDIR}/usr/share/applications/peazip.desktop"
+
+mkdir -v "${PKGDIR}/usr/bin" -p
+pushd "${PKGDIR}/usr/bin"
+ln -sv ../lib/peazip/peazip ./peazip
```
Comment (MingcongBai) [inline extra-utils/peazip/autobuild/build]: Save yourself some pushd... use this:

`ln -sv ../lib/peazip/peazip "$PKGDIR"/usr/bin/peazip`

File: extra-utils/peazip/autobuild/build [hunk 14]
```diff
@@ -0,0 +1,43 @@
...
+ln -sv ../lib/peazip/pea ./pea
```
Comment (MingcongBai) [inline extra-utils/peazip/autobuild/build]: Same here.

File: extra-utils/peazip/autobuild/build [hunk 15]
```diff
@@ -0,0 +1,43 @@
...
+popd
+
+pushd "${PKGDIR}/usr/lib/peazip"
+for i in 7z brotli zpaq zstd upx unrar unace
+do
+  mkdir -v "${PKGDIR}/usr/lib/peazip/res/$i" -p
+  ln -sv ../../../../bin/$i ./res/$i/$i
```
Comment (MingcongBai) [inline extra-utils/peazip/autobuild/build]: As above.

File: extra-utils/peazip/autobuild/build [hunk 16]
```diff
@@ -0,0 +1,43 @@
...
+done
+popd
+
+abinfo "Installing PeaZip L10n Data..."
+install -v -Dm644 "${SRCDIR}/../peazip-${PKGVER}.about_translations/lang"/*.txt -t "${PKGDIR}/usr/lib/peazip/res/lang"
+
+abinfo "Installing KDE's Dolphin Integrations..."
+install -v -Dm644 "${SRCDIR}/FreeDesktop_integration/kde4-dolphin/usr/share/kde4/services/ServiceMenus"/*.desktop -t "${PKGDIR}/usr/share/kservices5/ServiceMenus"
```
Comment (MingcongBai) [inline extra-utils/peazip/autobuild/build]: For long commands, its best to wrap lines.

**Review by MingcongBai — CHANGES_REQUESTED** [review]

**Review by MingcongBai — APPROVED** [review]

**Pull request merged**